server/neptune/workflows/internal/deploy/revision/queue/revision_processor.go

```diff
@@ -0,0 +1,100 @@
+package queue
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/root"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+type githubActivities interface {
+	CompareCommit(ctx context.Context, request activities.CompareCommitRequest) (activities.CompareCommitResponse, error)
+	UpdateCheckRun(ctx context.Context, request activities.UpdateCheckRunRequest) (activities.UpdateCheckRunResponse, error)
+}
+
+type RevisionProcessor struct {
+	Activities githubActivities
+}
+
+const (
+	DirectionBehindSummary   = "This revision is behind the current revision and will not be deployed.  If this is intentional, revert the default branch to this revision to trigger a new deployment."
+	UpdateCheckRunRetryCount = 5
+
+	DivergedCommitsSummary = "The current deployment has diverged from the default branch, so we have locked the root. This is most likely the result of this PR performing a manual deployment. To override that lock and allow the main branch to perform new deployments, select the Unlock button."
+)
+
+func (p *RevisionProcessor) Process(ctx workflow.Context, requestedDeployment terraform.DeploymentInfo, latestDeployment *root.DeploymentInfo) error {
+	commitDirection, err := p.getDeployRequestCommitDirection(ctx, requestedDeployment, latestDeployment)
+	if err != nil {
+		return err
+	}
+
+	switch commitDirection {
+	case activities.DirectionBehind:
+		// always returns error for caller to skip revision
+		if err = p.updateCheckRun(ctx, requestedDeployment, github.CheckRunFailure, DirectionBehindSummary, nil); err != nil {
+			logger.Error(ctx, "unable to update check run", err.Error())
+		}
+		return fmt.Errorf("requested revision %s is behind current one %s", requestedDeployment.Revision, latestDeployment.Revision)
```

Comment:
This is kind of weird that it's an error because when we log it we log as an error.  Errors should be things that are out of the ordinary, otherwise warns usually suffice.  I understand it makes the function a bit more complex to have to worry about that state but the way I'd go around it is just by moving the TerraformWorkflowRunner call in this revision processor.  I think this would also make the name make a bit more sense.

The worker just needs to know that the revision was processed.  Processing entails a validation component and then the actual processing which would be running the tf workflow.  Right now processor doesn't hold true to its name tbh.